crates/gwyneth/src/exex.rs

```diff
@@ -91,7 +96,13 @@ impl<Node: reth_node_api::FullNodeComponents> Rollup<Node> {
             }
 
             if let Some(committed_chain) = notification.committed_chain() {
-                self.commit(&committed_chain).await?;
+                let nodes = &self.nodes;
+                let engine_apis = &self.engine_apis;
+                for i in 0..nodes.len() {
+                    let node = &nodes[i];
```

Comment:
maybe easy solution: pass in the node idx into commit to avoid having to borrow here and just get the node in commit instead